rag/core/ingest_handler.py

```diff
@@ -199,15 +228,37 @@ def ingest_file(self, file_data: bytes, metadata: Dict[str, Any]) -> Dict[str, A
         file_size = len(file_data)
         logger.info(f"Ingesting file: {filename}")
 
-        # Log file processing start
-        self.process_logger.log_file_processing(
-            filename,
-            "started",
-            {
-                "size_bytes": file_size,
-                "content_type": metadata.get("content_type", "unknown"),
-            },
-        )
+        # Initialize event logger if observability is available
+        event_logger = None
+        if EventLogger and hash_config and save_config_snapshot:
+            try:
+                # Load config and hash it
+                config = load_config(str(self.config_path))
```

Comment:
Load the config in the constructor so that you can do it once for any function in here that needs it and also get the namespace and project ID in a more robust manner